File: pkg/deb/Makefile.php
MODULES+=		php
MODULE_SUFFIX_php=	php

MODULE_SUMMARY_php=	PHP module for $(BRAND_TITLE)

MODULE_VERSION_php=	$(VERSION)
MODULE_RELEASE_php=	1

MODULE_CONFARGS_php=	php
MODULE_MAKEARGS_php=	php
MODULE_INSTARGS_php=	php-install

MODULE_SOURCES_php=	freeunit.example-php-app \
			freeunit.example-php-config

ifneq (,$(findstring $(CODENAME),trusty jessie))
BUILD_DEPENDS_php=	php5-dev libphp5-embed
MODULE_BUILD_DEPENDS_php=,php5-dev,libphp5-embed
MODULE_DEPENDS_php=,libphp5-embed
else
BUILD_DEPENDS_php=	php-dev libphp-embed
MODULE_BUILD_DEPENDS_php=,php-dev,libphp-embed
MODULE_DEPENDS_php=,libphp-embed
endif

BUILD_DEPENDS+=		$(BUILD_DEPENDS_php)

define MODULE_PREINSTALL_php
	mkdir -p debian/$(BRAND)-php/usr/share/doc/$(BRAND)-php/examples/phpinfo-app
	install -m 644 -p debian/freeunit.example-php-app debian/$(BRAND)-php/usr/share/doc/$(BRAND)-php/examples/phpinfo-app/index.php
	install -m 644 -p debian/freeunit.example-php-config debian/$(BRAND)-php/usr/share/doc/$(BRAND)-php/examples/$(BRAND).config
endef
export MODULE_PREINSTALL_php

define MODULE_POST_php
cat <<BANNER
----------------------------------------------------------------------

The $(MODULE_SUMMARY_php) has been installed.

To check out the sample app, run these commands:

 sudo service $(RUNTIME) restart
 cd /usr/share/doc/$(BRAND)-$(MODULE_SUFFIX_php)/examples
 $(MODULE_CONFIG_PUT)
 curl http://localhost:8300/

Online documentation is available at $(DOCS_URL)

----------------------------------------------------------------------
BANNER
endef
export MODULE_POST_php

File: pkg/deb/Makefile.go
MODULES+=		go
MODULE_SUFFIX_go=	go

MODULE_SUMMARY_go=	Go module for $(BRAND_TITLE)

MODULE_VERSION_go=	$(VERSION)
MODULE_RELEASE_go=	1

MODULE_CONFARGS_go=	go --go-path=/usr/share/gocode
MODULE_MAKEARGS_go=	go
MODULE_INSTARGS_go=	go-install-src

MODULE_SOURCES_go=	freeunit.example-go-app \
			freeunit.example-go-config

BUILD_DEPENDS_go=	golang
BUILD_DEPENDS+=		$(BUILD_DEPENDS_go)

MODULE_BUILD_DEPENDS_go=,golang
MODULE_DEPENDS_go=,golang,$(BRAND)-dev (= $(VERSION)-$(RELEASE)~$(CODENAME))

MODULE_NOARCH_go=	true

define MODULE_PREINSTALL_go
	mkdir -p debian/$(BRAND)-go/usr/share/doc/$(BRAND)-go/examples/go-app
	install -m 644 -p debian/freeunit.example-go-app debian/$(BRAND)-go/usr/share/doc/$(BRAND)-go/examples/go-app/let-my-people.go
	install -m 644 -p debian/freeunit.example-go-config debian/$(BRAND)-go/usr/share/doc/$(BRAND)-go/examples/$(BRAND).config
endef
export MODULE_PREINSTALL_go

define MODULE_POST_go
cat <<BANNER
----------------------------------------------------------------------

The $(MODULE_SUMMARY_go) has been installed.

To check out the sample app, run these commands:

 GOPATH=/usr/share/gocode GO111MODULE=auto go build -o /tmp/go-app /usr/share/doc/$(BRAND)-$(MODULE_SUFFIX_go)/examples/go-app/let-my-people.go
 sudo service $(RUNTIME) restart
 cd /usr/share/doc/$(BRAND)-$(MODULE_SUFFIX_go)/examples
 $(MODULE_CONFIG_PUT)
 curl http://localhost:8500/

Online documentation is available at $(DOCS_URL)

----------------------------------------------------------------------
BANNER
endef
export MODULE_POST_go
